src/core/api/media_element_track_choice_manager.ts

```diff
@@ -0,0 +1,468 @@
+        return track;
+      }
+    }
+    return null;
+  }
+
+  public getAvailableAudioTracks(): ITMAudioTrackListItem[]|undefined {
+    return this._audioTracks.map(({ track, origTrack }) => {
+      return { id: track.id,
+               language: track.language,
+               normalized: track.normalized,
+               audioDescription: track.audioDescription,
+               active: origTrack.enabled };
+    });
+  }
+
+  public getAvailableTextTracks(): ITMTextTrackListItem[]|undefined {
+    return this._textTracks.map(({ track, origTrack }) => {
+      return { id: track.id,
+               language: track.language,
+               normalized: track.normalized,
+               closedCaption: track.closedCaption,
+               active: origTrack.mode === "showing" };
+    });
+  }
+
+  public getAvailableVideoTracks(): ITMVideoTrackListItem[]|undefined {
+    return this._videoTracks.map(({ track, origTrack }) => {
+      return { id: track.id,
+               representations: track.representations,
+               active: origTrack.selected };
+    });
+  }
+
+  public dispose(): void {
+    this._mediaElement.videoTracks.onchange = null;
+    this._mediaElement.videoTracks.onaddtrack = null;
+    this._mediaElement.videoTracks.onremovetrack = null;
+
+    this._mediaElement.audioTracks.onchange = null;
+    this._mediaElement.audioTracks.onaddtrack = null;
+    this._mediaElement.audioTracks.onremovetrack = null;
+
+    (this._mediaElement.textTracks as ICustomTextTrackList).onchange = null;
+    this._mediaElement.textTracks.onaddtrack = null;
+    (this._mediaElement.textTracks as ICustomTextTrackList).onremovetrack = null;
+
+    this.removeEventListener();
+  }
+
+  private _setInitialAudioTrack() : void {
+    const preferredAudioTracks = this._preferredAudioTracks.getValue();
+    const normalizedLanguages = preferredAudioTracks
+      .filter(
+        (audioTrack): audioTrack is {
+          language : string;
+          audioDescription : boolean;
+        } => audioTrack !== null)
+      .map(({ language }) => normalizeLanguage(language));
+
+    for (let i = 0; i < normalizedLanguages.length; i++) {
+      const language = normalizedLanguages[i];
+      for (let j = 0; j < this._audioTracks.length; j++) {
+        const audioTrack = this._audioTracks[j];
+        const normalizedLanguage = audioTrack.track.normalized;
+        if (normalizedLanguage === language) {
+          this.setAudioTrackById(audioTrack.track.id);
+          return;
+        }
+      }
+    }
+  }
+
+  private _setInitialTextTrack() : void {
+    const preferredTextTracks = this._preferredTextTracks.getValue();
+    const normalizedLanguages = preferredTextTracks
+      .filter(
+        (textTrack): textTrack is { language : string;
+                                    closedCaption : boolean; } => textTrack !== null)
+      .map(({ language }) =>  normalizeLanguage(language));
+
+    for (let i = 0; i < normalizedLanguages.length; i++) {
```

Comment:
`closedCaption` does exist on the `kind` attribute of a text track.

Reply:
done